framework/integration/reconciliation_test.go

```diff
@@ -0,0 +1,63 @@
+// +build k8srequired
+
+package integration
+
+import (
+	"testing"
+	"time"
+
+	"github.com/giantswarm/operatorkit/informer"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Test_Finalizer_Integration_Reconciliation is a integration test for basic finalizer
+// operations. The test verifies that finalizers are added and removed as
+// expected. It does not cover correct behavior with reconciliation.
+func Test_Finalizer_Integration_Reconciliation(t *testing.T) {
+	namespace := "finalizer-integration-reconciliation-test"
+	mustSetup(namespace)
+	defer mustTeardown(namespace)
+
+	operatorName := "test-operator"
+	configMapName := "test-cm"
+
+	c := informer.Config{
+		Watcher: k8sClient.CoreV1().ConfigMaps(namespace),
+
+		RateWait:     time.Second * 2,
+		ResyncPeriod: time.Second * 10,
+	}
+	operatorkitInformer, err := informer.New(c)
+	if err != nil {
+		t.Fatal("expected", nil, "got", err)
+	}
+
+	operatorkitFramework, err := newFramework(operatorName, namespace, operatorkitInformer)
+	if err != nil {
+		t.Fatal("expected", nil, "got", err)
+	}
+	operatorkitFramework.Boot()
```

Comment:
This sets `context.TODO()` at the very start which is then blocking apparently. The integration test for informers used `context.Background()`.

Reply:
So the output I see when running this locally is:
```
{"caller":"github.com/giantswarm/operatorkit/framework/framework.go:230","event":"update","function":"UpdateFunc","level":"debug","message":"stop framework reconciliation due to finalizer added","time":"2018-03-27 14:24:01.446"}
{"caller":"github.com/giantswarm/operatorkit/framework/framework.go:271","function":"bootWithError","level":"debug","message":"starting list-watch","time":"2018-03-27 14:24:01.825"}
```
Then it's just the framework waiting for new events.